include/FeedingFanOut.h

```diff
@@ -148,7 +152,7 @@ namespace ChimeraTK {
 
   template<typename UserType>
   void FeedingFanOut<UserType>::doReadTransferSynchronously() {
-    if(this->_disabled) {
+    if(this->_disabled || !_hasReturnSlave) {
```

Comment:
Do you need the _hadReturnSlave variable at all? You could just check whether the shared pointer is set.